fv3core/stencils/d_sw.py

```diff
@@ -114,75 +180,92 @@ def pressure_and_vbke(
     uc: FloatField,
     cosa: FloatFieldIJ,
     rsina: FloatFieldIJ,
-    vt: FloatField,
-    vb: FloatField,
-    dt4: float,
-    dt5: float,
+    v: FloatField,
+    vc_contra: FloatField,
+    u: FloatField,
+    uc_contra: FloatField,
+    ke: FloatField,
+    dx: FloatFieldIJ,
+    dxa: FloatFieldIJ,
+    rdx: FloatFieldIJ,
+    dy: FloatFieldIJ,
+    dya: FloatFieldIJ,
+    rdy: FloatFieldIJ,
+    dt: float,
 ):
-    from __externals__ import inline_q, local_ie, local_is, local_je, local_js
 
     with computation(PARALLEL), interval(...):
-        # TODO: only needed for d_sw validation
-        if __INLINED(inline_q == 0):
-            with horizontal(region[local_is : local_ie + 1, local_js : local_je + 1]):
-                pt = flux_integral(pt, delp, gx, gy, rarea)
-                delp = delp + flux_component(fx, fy, rarea)
-                pt = pt / delp
-        vb = vbke(vc, uc, cosa, rsina, vt, vb, dt4, dt5)
-
-
-@gtscript.function
-def ke_from_bwind(ke, ub, vb):
-    return 0.5 * (ke + ub * vb)
+        pt, delp = pressure_update(gx, gy, rarea, fx, fy, pt, delp)
+        # vb_contra = interpolate_vc_to_cell_corners(vc, uc, cosa, rsina, vt)
+        ub_contra, vb_contra = interpolate_uc_vc_to_cell_corners(
+            uc, vc, cosa, rsina, uc_contra, vc_contra
+        )
+        advected_v = advect_v_along_y(vb_contra, v, dy, dya, rdy, dt)
+        advected_u = advect_u_along_x(ub_contra, u, dx, dxa, rdx, dt)
+        # TODO: we see here that ke is not kinetic energy, but kinetic energy * timestep
+        #       refactor or rename to avoid this confusion
+        ke = 0.5 * dt * (ub_contra * advected_u + vb_contra * advected_v)
+        ke = all_corners_ke(ke, u, v, uc_contra, vc_contra, dt)
 
 
-def ub_vb_from_vort(
+def vort_differencing_stencil_defn(
     vort: FloatField,
-    ub: FloatField,
-    vb: FloatField,
+    vort_x_delta: FloatField,
+    vort_y_delta: FloatField,
     dcon: FloatFieldK,
 ):
+    with computation(PARALLEL), interval(...):
+        vort_x_delta, vort_y_delta = vort_differencing(vort, dcon)
+
+
+@gtscript.function
+def vort_differencing(vort, dcon):
     from __externals__ import local_ie, local_is, local_je, local_js
 
-    with computation(PARALLEL), interval(...):
-        if dcon[0] > dcon_threshold:
-            # Creating a gtscript function for the ub/vb computation
-            # results in an "NotImplementedError" error for Jenkins
-            # Inlining the ub/vb computation in this stencil resolves the Jenkins error
-            with horizontal(region[local_is : local_ie + 1, local_js : local_je + 2]):
-                ub = vort - vort[1, 0, 0]
-            with horizontal(region[local_is : local_ie + 2, local_js : local_je + 1]):
-                vb = vort - vort[0, 1, 0]
+    # TODO: Why is this if statement here?
+    # Can/should it be moved outside this function?
```

Comment:
This doesn't need to be a function, it's only used once, might as well put it in the stencil and not add a TODO.